--- ru	2022-11-21 13:54:40.710682785 +0300
+++ ru	2022-11-22 08:37:38.853069692 +0300
@@ -863,18 +863,30 @@
     include "ru(common)"
     name[Group1]= "Russian (Macintosh)";
 
-    key <LSGT> { [          less,      greater  ] };
-    key <AE02> { [             2,     quotedbl  ] };
-    key <AE03> { [             3,   numerosign  ] };
-    key <AE04> { [             4,      percent  ] };
-    key <AE05> { [             5,        colon  ] };
-    key <AE06> { [             6,        comma  ] };
-    key <AE07> { [             7,       period  ] };
-    key <AE08> { [             8,    semicolon  ] };
-    key <TLDE> { [  bracketright,  bracketleft  ] };
-    key <BKSL> { [   Cyrillic_io,  Cyrillic_IO  ] };
+    key <LSGT> { [          less,      greater                                     ] };
+    key <AE01> { [             1,       exclam,             exclam,      trademark ] };
+    key <AE02> { [             2,     quotedbl,                 at,        section ] };
+    key <AE03> { [             3,   numerosign,         numbersign,       sterling ] };
+    key <AE04> { [             4,    semicolon,             dollar,       EuroSign ] };
+    key <AE05> { [             5,      percent,          copyright,     registered ] };
+    key <AE06> { [             6,        colon,        asciicircum,        notsign ] };
+    key <AE07> { [             7,     question,          ampersand,      paragraph ] };
+    key <AE08> { [             8,     asterisk,              U20BD,         degree ] };
+    key <AE09> { [             9,    parenleft,          parenleft,      braceleft ] };
+    key <AE10> { [             0,   parenright,         parenright,     braceright ] };
+    key <AE11> { [         minus,   underscore,             endash,         emdash ] };
+    key <AE12> { [         equal,         plus,           notequal,          U2248 ] };
+    key <TLDE> { [  bracketright,  bracketleft                                     ] };
+    key <AB10> { [        period,        comma                                     ] };
+    key <BKSL> { [     backslash,        slash,          backslash,            bar ] };
 
-    key <AC06> { [   Cyrillic_er,  Cyrillic_ER, U20BD ] };
+    key <AC04> { [   Cyrillic_a,    Cyrillic_A,           division,      plusminus ] };
+    key <AC05> { [   Cyrillic_pe,  Cyrillic_PE,           ellipsis,     squareroot ] };
+    key <AC06> { [   Cyrillic_er,  Cyrillic_ER, enfilledcircbullet,       infinity ] };
+    key <AC10> { [  Cyrillic_zhe, Cyrillic_ZHE,      guillemotleft, guillemotright ] };
+
+    key <AB08> { [   Cyrillic_be,  Cyrillic_BE,      lessthanequal,           less ] };
+    key <AB09> { [   Cyrillic_yu,  Cyrillic_YU,   greaterthanequal,        greater ] };
 
     include "level3(ralt_switch)"
 };
