--- ru	2022-11-21 13:54:40.710682785 +0300
+++ ru	2023-08-14 17:10:21.527682209 +0300
@@ -863,18 +863,32 @@
     include "ru(common)"
     name[Group1]= "Russian (Macintosh)";
 
-    key <LSGT> { [          less,      greater  ] };
-    key <AE02> { [             2,     quotedbl  ] };
-    key <AE03> { [             3,   numerosign  ] };
-    key <AE04> { [             4,      percent  ] };
-    key <AE05> { [             5,        colon  ] };
-    key <AE06> { [             6,        comma  ] };
-    key <AE07> { [             7,       period  ] };
-    key <AE08> { [             8,    semicolon  ] };
-    key <TLDE> { [  bracketright,  bracketleft  ] };
-    key <BKSL> { [   Cyrillic_io,  Cyrillic_IO  ] };
-
-    key <AC06> { [   Cyrillic_er,  Cyrillic_ER, U20BD ] };
+    key <LSGT> { [          less,      greater                                     ] };
+    key <AE01> { [             1,       exclam,             exclam,      trademark ] };
+    key <AE02> { [             2,     quotedbl,                 at,        section ] };
+    key <AE03> { [             3,   numerosign,         numbersign,       sterling ] };
+    key <AE04> { [             4,    semicolon,             dollar,       EuroSign ] };
+    key <AE05> { [             5,      percent,          copyright,     registered ] };
+    key <AE06> { [             6,        colon,        asciicircum,        notsign ] };
+    key <AE07> { [             7,     question,          ampersand,      paragraph ] };
+    key <AE08> { [             8,     asterisk,              U20BD,         degree ] }; // 8, *, rouble sign, degree sign
+    key <AE09> { [             9,    parenleft,          parenleft,      braceleft ] };
+    key <AE10> { [             0,   parenright,         parenright,     braceright ] };
+    key <AE11> { [         minus,   underscore,             endash,         emdash ] };
+    key <AE12> { [         equal,         plus,           notequal,          U2248 ] }; // =, -, notequal, almost equal
+    key <TLDE> { [  bracketright,  bracketleft                                     ] };
+    key <AB10> { [        period,        comma                                     ] };
+    key <BKSL> { [     backslash,        slash,          backslash,            bar ] };
+
+    key <AC04> { [   Cyrillic_a,    Cyrillic_A,           division,      plusminus ] };
+    key <AC05> { [   Cyrillic_pe,  Cyrillic_PE,           ellipsis,     squareroot ] };
+    key <AC06> { [   Cyrillic_er,  Cyrillic_ER, enfilledcircbullet,       infinity ] };
+    key <AC07> { [    Cyrillic_o,   Cyrillic_O,              U2206,       Greek_mu ] };  // о, О, increment, Greek mu
+    key <AC10> { [  Cyrillic_zhe, Cyrillic_ZHE,      guillemotleft, guillemotright ] };
+
+    key <AB06> { [   Cyrillic_te,  Cyrillic_TE,           function,       function ] };
+    key <AB08> { [   Cyrillic_be,  Cyrillic_BE,      lessthanequal,           less ] };
+    key <AB09> { [   Cyrillic_yu,  Cyrillic_YU,   greaterthanequal,        greater ] };
 
     include "level3(ralt_switch)"
 };
@@ -1056,3 +1070,21 @@
 
     include "level3(ralt_switch)"
 };
+
+// Apple layout with Alexey Martynov's changes
+partial alphanumeric_keys
+xkb_symbols "mac_avm" {
+
+    include "ru(mac)"
+    name[Group1]= "Russian (Macintosh, Alexey Martynov)";
+
+    key <SPCE> { [         space,        space,       nobreakspace,     0x100200D ] };  // space, space, nobreakspace, zero width joiner
+
+    key <AD03> { [    Cyrillic_u,   Cyrillic_U,      Greek_epsilon,         U0301 ] };  // у, У, Greek epsilon, combining acute
+    key <AD07> { [  Cyrillic_ghe, Cyrillic_GHE,       Cyrillic_ghe,         U0308 ] };  // г, Г, г, combining diaeresis
+
+    key <AC09> { [   Cyrillic_de,  Cyrillic_DE,       Greek_lambda                ] };
+
+    include "level3(ralt_switch)"
+};
+
