--- us	2022-12-07 11:52:53.533386974 +0300
+++ us	2023-08-14 17:07:10.411388578 +0300
@@ -2162,3 +2162,15 @@
 
     include "level3(ralt_switch)"
 };
+
+partial alphanumeric_keys
+xkb_symbols "mac_avm" {
+
+    include "us(mac)"
+    name[Group1]= "English (Macintosh, Alexey Martynov)";
+
+    key <SPCE> { [         space,        space,       nobreakspace,      0x100200D ] };  // space, space, nobreakspace, zero width joiner
+
+    key <AD03> { [	        e,           E,         dead_acute,          U0301 ] };  // e, E, acute (dead keys), combining acute
+    key <AD07> { [	        u,           U,     dead_diaeresis,          U0308 ] };  // u, U, diaeresis (dead keys), combining diaeresis
+};
